packages/frontend/src/stores/messagelist.ts

```diff
@@ -53,11 +53,12 @@ const defaultState = () =>
   }) as MessageListState
 
 /*
- * A hook to read a portion of messages(a view) for a given chat. It creates a store(MessageListStore)
+ * A hook to read a portion of messages(a view) for a given chat. It uses the passed store(MessageListStore)
  * for the given chat and account and loads messages on it. It always has a maximum specified number
- * of messages as per PAGE_SIZE constant.
+ * of messages as per PAGE_SIZE constant. The store is also used in useDraft to avoid redundant data loading.
```

Comment:
All my other comments from this review aside, it doesn't seem to me that trying to decouple `useMessageList` from `MessageListStore` is good idea. What is the point of it if `useMessageList` is the only thing that is supposed to be responsible for managing (mutating) the `MessageListStore`? Basically `MessageListStore` can never exist without the corresponding `useMessageList`. But then why would we spread the creation of `MessageListStore` and the invocation of `useMessageList` so far apart?

I suggest to revert the decoupling, and to simply lift the `useMessageList()` call to inside `MessageListAndComposer`, where `useDraft` can access the return value of `useMessageList()`.
With this done, I believe all my other comments from this review will become irrelevant.